Sources/VLCTranscoder.m

```diff
@@ -0,0 +1,142 @@
+/*****************************************************************************
+ * VLCTranscoder.m: VLCKit.framework VLCTranscoder implementation
+ *****************************************************************************
+ * Copyright (C) 2018 Carola Nitz
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors:  Carola Nitz <caro # videolan.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+#import "VLCTranscoder.h"
+
+#import "VLCEventManager.h"
+#import "VLCLibrary.h"
+#import "VLCLibVLCBridging.h"
+
+#include <vlc/vlc.h>
+
+@interface VLCTranscoder()
+{
+    libvlc_media_player_t *_p_mp; //player instance used for transcoding
+    dispatch_queue_t _libVLCTranscoderQueue;
+}
+- (void)registerObserversForMuxWithPlayer:(libvlc_media_player_t *)player;
+- (void)unregisterObserversForMuxWithPlayer:(libvlc_media_player_t *)player;
+@end
+
+@implementation VLCTranscoder
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        _libVLCTranscoderQueue = dispatch_queue_create("libVLCTranscoderQueue", DISPATCH_QUEUE_SERIAL);
+    }
+    return self;
+}
+
+- (BOOL)muxSubtitleFile:(NSString *)srtPath toMp4File:(NSString *)mp4Path outputPath:(NSString *)outputPath
+{
+    libvlc_media_t* p_media = libvlc_media_new_path([[VLCLibrary sharedLibrary] instance], [mp4Path UTF8String]);
+    if (p_media == NULL) {
+        NSAssert(0, @"p_media wasn't allocated");
+        return NO;
+    }
+    NSString *transcodingOptions = [NSString stringWithFormat:@":sout=#transcode{venc={module=avcodec{codec=h264_videotoolbox}, vcodec=h264},venc={module=vpx{quality-mode=2},vcodec=VP80},acodec=mpga,ab=128,channels=2,samplerate=44100,soverlay}:file{dst='%@',mux=mkv}", outputPath];
```

Comment:
Not sure if hardcoding the audio codec to MPGA with only 128 kbits is a good idea? Doesn't that sound awful?

Reply:
tbh this part of the code is not my area of expertise so can I just drop it?

Reply:
Ideally you would leave the audio alone here, that said depending on the source the audio codecs could maybe be incompatible with the MKV container.

Reply:
so from speaking to @alexandre-janniaux I can't drop the audio options here

Reply:
So VLC has no way, like FFMpeg has, to "copy" a specific track to the output without transcoding it?

Reply:
@alexandre-janniaux can you answer @ePirat because this is out of my knowledge scope

Reply:
There's no way for an elementary stream to avoid the transcode module but removing the audio parameters should tell the transcode module to avoid transcoding them and use them as-is.

I don't know the possible timestamp issue that could be raised by that, but it should be ok.

Reply:
I can try leaving it out